front/components/vaults/VaultDataSourceViewContentList.tsx

```diff
@@ -98,27 +132,99 @@ export const VaultDataSourceViewContentList = ({
             : ""
         )}
       >
+        <DropdownMenu>
+          <DropdownMenu.Button>
+            <Button
+              size="sm"
+              label="Add data"
+              icon={PlusIcon}
+              variant="primary"
+              type="menu"
+            />
+          </DropdownMenu.Button>
+
+          <DropdownMenu.Items width={300}>
+            <DropdownMenu.Item
+              icon={DocumentTextIcon}
+              onClick={() => {
+                contentActionsRef.current?.callAction(
+                  "DocumentUploadOrEditModal"
+                );
+              }}
+              label={"Create a document"}
+            />
+            <DropdownMenu.Item
+              icon={TableIcon}
+              onClick={() => {
+                contentActionsRef.current?.callAction("TableUploadOrEditModal");
+              }}
+              label={"Create a table"}
+            />
+            <DropdownMenu.Item
+              icon={CloudArrowUpIcon}
+              onClick={() => {
+                contentActionsRef.current?.callAction(
+                  "MultipleDocumentsUpload"
+                );
+              }}
+              label={"Upload multiple files"}
+            />
+          </DropdownMenu.Items>
+        </DropdownMenu>
         {rows.length > 0 && (
-          <Searchbar
-            name="search"
-            placeholder="Search (Name)"
-            value={dataSourceSearch}
-            onChange={(s) => {
-              setDataSourceSearch(s);
-            }}
-          />
+          <>
+            <Searchbar
+              name="search"
+              placeholder="Search (Name)"
+              value={dataSourceSearch}
+              onChange={(s) => {
+                setDataSourceSearch(s);
+              }}
+            />
+            <DropdownMenu>
```

Comment:
This requires more logic actually since we want to apply a facet to the local data.